account_billing/models/account_billing.py

```diff
@@ -0,0 +1,194 @@
+        default=lambda self: self._context.get('type', False)
+    )
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form',
+                        toolbar=False, submenu=False):
+        res = super(AccountBilling, self).fields_view_get(
+            view_id=view_id, view_type=view_type,
+            toolbar=toolbar, submenu=submenu)
+        active_ids = self._context.get('active_ids', False)
+        invoice_ids = self.env['account.invoice'].search([
+            ('id', 'in', active_ids)])
+        billing_ids = invoice_ids.mapped('billing_id')
+        if any(inv.state != 'open' for inv in invoice_ids):
+            raise ValidationError(_("The billing cannot be processed \
+                because the invoice is not open!"))
+        if billing_ids:
+            raise ValidationError(_("Invoice(s) create billing already!"))
+        return res
+
+    @api.onchange('type')
+    def _onchange_partner_id(self):
+        self.ensure_one()
+        # Set partner_id domain
+        if self.type:
+            return {'domain': {'partner_id': [
+                (MAP_INVOICE_TYPE_PARTNER_TYPE[self.type], '=', True)]}}
+
+    @api.multi
+    def _get_partner_id(self):
+        active_ids = self._context.get('active_ids', False)
+        if active_ids:
+            partner_ids = self.env['account.invoice'].search([
+                ('id', 'in', active_ids)]).mapped('partner_id')
+            if len(partner_ids) > 1:
+                raise ValidationError(_("Please select the same partner name"))
+            return partner_ids
+        return False
+
+    @api.multi
+    def _compute_type(self):
+        for rec in self:
+            rec.type = rec._context.get('type', False)
+
+    @api.multi
+    @api.depends('invoice_ids')
+    def _compute_invoice_related_count(self):
+        for billing in self:
+            invoice_ids = self.env['account.invoice'].search([
+                ('id', 'in', billing.invoice_ids.ids)
+            ])
+            billing.invoice_related_count = len(invoice_ids)
+
+    @api.multi
+    def name_get(self):
+        result = []
+        for billing in self:
+            result.append((billing.id, (billing.number or 'Draft')))
+        return result
+
+    @api.multi
+    def validate_billing(self):
+        for rec in self:
+            if any(inv.state != 'open' for inv in rec.invoice_ids):
+                raise ValidationError(_("The billing cannot be processed \
+                    because the invoice is not open!"))
+            # keep the number in case of a billing reset to draft
+            if not rec.number:
+                # Use the right sequence to set the name
+                if rec.type == 'out_invoice':
+                    sequence_code = 'account.customer.billing'
+                if rec.type == 'in_invoice':
+                    sequence_code = 'account.supplier.billing'
+                rec.number = \
+                    self.env['ir.sequence'].next_by_code(sequence_code)
```

Comment:
Make sure it will work with fiscal year.
```suggestion
                    self.env['ir.sequence']..with_context(ir_sequence_date=rec.payment_date).next_by_code(sequence_code)
```